tembo-py/README.md

```diff
@@ -2,11 +2,145 @@
 
 The official Python client for Tembo.io
 
+For more technical information and ways to get involved, please refer to the [contributing guide](./CONTRIBUTING.md).
+
+## Prerequisites
+
+- [Python](https://www.python.org/) - Programming language and companion `pip` package manager
+- [Docker Engine](https://docs.docker.com/engine/install/) - For running local containers
 
 ## Installation
 
+The [tembo-py library](https://pypi.org/project/tembo-py/) is hosted on pypi.org and can be installed using the following `pip` command.
+
 ```bash
 pip install tembo-py
 ```
 
+## Example: Adding Custom Prompts
+
+Within the [rag.py](./tembo_py/rag.py) file's `TemboRAG` class, the `add_prompt_template` method introduces the ability to add custom prompts.
+For the purposes of this example we will consider a local environment.
+
+#### 1. Running PostgreSQL
+
+Before even touching Python, let's go ahead and start a Postgres instance.
+If you'd like, you can run the following to clone and enter the `tembo-py` repository:
+
+```bash
+git clone https://github.com/tembo-io/tembo.git
+```
+```bash
+cd tembo/tembo-py
+```
+
+Then run the following command to start a Postgres instance:
+
+```bash
+make run.postgres
+```
+
+From there, `psql` into Postgres and enable the [pg_vectorize](https://github.com/tembo-io/pg_vectorize) extension.
+
+:bulb: Note that the password is by default `postgres`.
+
+```bash
+psql -h localhost -p 5432 -U postgres -W
+```
+
+```sql
+CREATE EXTENSION vectorize CASCADE;
+```
+
+#### 2. Create a New Python File
+
+Exit Postgres and create a new Python file:
+
+```bash
+touch example_tembo.py
+```
+
+#### 3. Define Your Custom Prompt
 
+Using your preferred text editor or IDE, you can create the following script:
+
+```python
+from tembo_py.rag import TemboRAG
+import psycopg
+
+# Constants
+PROJECT_NAME = "test_add_prompt_template"
+PROMPT_NAME = "example_prompt"
+SYS_PROMPT = "System prompt text for example"
+USER_PROMPT = "User prompt text for example"
+CONNECTION_STRING = "postgresql://postgres:postgres@localhost:5432/postgres"
+
+def initialize_tembo_rag(project_name, connection_string):
+    """
+    Initialize TemboRAG instance.
+    """
+    return TemboRAG(project_name=project_name, connection_string=connection_string)
+
+def add_prompt_template_to_db():
+    # Initialize TemboRAG instance
+    tembo_rag = initialize_tembo_rag(PROJECT_NAME, CONNECTION_STRING)
+
+    # Add the prompt template
+    tembo_rag.add_prompt_template(PROMPT_NAME, SYS_PROMPT, USER_PROMPT)
```

Comment:
I think we can shrink this down into just a few steps.

```py
tembo_rag = TemboRAG(project_name=project_name, connection_string=connection_string)

tembo_rag.add_prompt_template(PROMPT_NAME, SYS_PROMPT, USER_PROMPT)
```

Then add the code snips from here: https://tembo.io/docs/tembo-stacks/rag

Then call `tembo_rag.query()`